airflow/contrib/hooks/gcp_dataproc_hook.py

```diff
@@ -141,24 +141,87 @@ def build(self):
         return self.job
 
 
+class _DataProcOperation(LoggingMixin):
+    """Continuously polls Dataproc Operation until it completes."""
+    def __init__(self, dataproc_api, operation):
+        self.dataproc_api = dataproc_api
+        self.operation = operation
+        self.operation_name = self.operation['name']
+
+    def wait_for_done(self):
+        if self._check_done():
+            return True
+
+        self.log.info(
+            'Waiting for Dataproc Operation %s to finish', self.operation_name)
+        while True:
+            time.sleep(10)
+            self.operation = (
+                self.dataproc_api.projects()
+                .regions()
+                .operations()
+                .get(name=self.operation_name)
+                .execute(num_retries=5))
+
+            if self._check_done():
+                return True
+
+    def get(self):
+        return self.operation
+
+    def _check_done(self):
+        if 'done' in self.operation:
+            if 'error' in self.operation:
+                self.log.warning(
+                    'Dataproc Operation %s failed with error: %s',
+                    self.operation_name, self.operation['error']['message'])
+                self._raise_error()
+            else:
+                self.log.info(
+                    'Dataproc Operation %s done', self.operation['name'])
+                return True
+        return False
+
+    def _raise_error(self):
+        raise Exception('Google Dataproc Operation %s failed: %s' %
+                        (self.operation_name, self.operation['error']['message']))
+
+
 class DataProcHook(GoogleCloudBaseHook):
+    """Hook for Google Cloud Dataproc APIs."""
     def __init__(self,
                  gcp_conn_id='google_cloud_default',
                  delegate_to=None):
         super(DataProcHook, self).__init__(gcp_conn_id, delegate_to)
 
     def get_conn(self):
-        """
-        Returns a Google Cloud DataProc service object.
-        """
+        """Returns a Google Cloud Dataproc service object."""
         http_authorized = self._authorize()
         return build('dataproc', 'v1', http=http_authorized)
 
     def submit(self, project_id, job, region='global'):
         submitted = _DataProcJob(self.get_conn(), project_id, job, region)
         if not submitted.wait_for_done():
-            submitted.raise_error("DataProcTask has errors")
+            submitted.raise_error('DataProcTask has errors')
 
     def create_job_template(self, task_id, cluster_name, job_type, properties):
-        return _DataProcJobBuilder(self.project_id, task_id, cluster_name, job_type,
-                                   properties)
+        return _DataProcJobBuilder(self.project_id, task_id, cluster_name,
+                                   job_type, properties)
+
+
+class DataProcBetaHook(GoogleCloudBaseHook):
```

Comment:
Is it possible to revise add a version param to DataprocHook instead of a new hook?